packages/api/src/express/chatRouter.ts

```diff
@@ -0,0 +1,97 @@
+import express from 'express';
+import { Experimental_StdioMCPTransport as StdioMCPTransport } from 'ai/mcp-stdio';
+import {
+  experimental_createMCPClient as createMCPClient,
+  InvalidToolArgumentsError,
+  NoSuchToolError,
+  ToolExecutionError,
+  streamText,
+} from 'ai';
+import { anthropic } from '@ai-sdk/anthropic';
+import { defaultLogger } from '@unchainedshop/logger';
+import rateLimit from 'express-rate-limit';
+
+// Define the rate limiter middleware
+const chatRateLimiter = rateLimit({
+  windowMs: 60 * 1000,
+  max: 5, // Limit each IP to 5 requests per windowMs
+  standardHeaders: true,
+  legacyHeaders: false,
+  message: {
+    error: 'Too many requests to /chat. Please wait and try again later.',
+  },
+});
+
+const chatRouter = express.Router();
+const { CHAT_API_PATH = '/chat', ANTHROPIC_API_KEY, ROOT_URL = 'http://localhost:4010' } = process.env;
+
+const rootUrl = ROOT_URL;
+try {
+  const url = new URL(rootUrl);
+  if (!['http:', 'https:'].includes(url.protocol))
+    throw new Error('Configuration error: Invalid protocol');
+} catch (error) {
+  defaultLogger.error('Invalid ROOT_URL environment variable');
+  throw error;
+}
+
+const errorHandler = (error: any): string => {
+  if (NoSuchToolError.isInstance(error)) return 'NoSuchToolError';
+  if (InvalidToolArgumentsError.isInstance(error)) return 'InvalidToolArgumentsError';
+  if (ToolExecutionError.isInstance(error)) return 'ToolExecutionError';
+  if (error?.message?.toLowerCase() === 'forbidden') return 'NetworkError';
+  defaultLogger.error(error);
+  return `Failed to stream response: ${error?.message || 'Unknown error'}`;
+};
+
+chatRouter.post(CHAT_API_PATH, chatRateLimiter, async (req: any, res) => {
+  const { messages } = req.body;
+  if (!ANTHROPIC_API_KEY) {
+    const logMessage =
+      'ANTHROPIC_API_KEY environment variable is not set. Please set it to use the chat API.';
+    defaultLogger.error(logMessage);
+    res.status(503).json({
+      error: {
+        code: 'CHAT_API_UNAVAILABLE',
+        message: 'Chat service temporarily unavailable. please try again letter',
+      },
+      id: null,
+    });
+    return;
+  }
+
+  let client;
+  try {
+    client = await createMCPClient({
+      transport: new StdioMCPTransport({
+        command: 'npx',
+        args: ['-y', 'supergateway', '--streamableHttp', `${rootUrl}/mcp`],
+        env: {},
+      }),
+    });
+
+    const tools = await client.tools();
+
+    const result = streamText({
+      system:
+        'do not include the data in your summary, just write a summary about it never list all the fields of a result, just summarize paragraph about your findings, if necessary',
+      model: anthropic('claude-4-sonnet-20250514'),
```

Comment:
I guess this should be defined in boot.ts of example, API could be model-independent then!